pkg/backend/httpstate/state.go

```diff
@@ -207,7 +207,7 @@ func (u *cloudUpdate) RecordAndDisplayEvents(
 		go recordEngineEvent(e, eventIdx)
 
 		if e.Type == engine.CancelEvent {
-			return
+			break
```

Comment:
Minor nit, and just an idea. If you changed this to a `defer wg.Wait()` right after the declaration of the `WaitGroup`, this sort of mistake could be avoided in the future. TBH, I dislike synchronization in destructors almost as much as I despise early `return`s, so feel free to decline -- it's just that we are already employing both patterns in this function already (the `close`ing of the channel in the earlier `defer`, for instance).